### PR TITLE
Added test for sosreport

### PR DESCRIPTION
**Pull request by Pavithra1602**
sosreport generates an archive of configuration and diagnostic information from the running system.

Signed-off-by: pavrampu@linux.vnet.ibm.com <pavrampu@in.ibm.com>

### CODE REVIEW

File: generic/sosreport.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        dist = distro.detect()
+        sm = SoftwareManager()
+        if dist.name == 'Ubuntu':
+            if not sm.check_installed("sosreport") and not sm.install("sosreport"):
+                self.skip("Fail to install sosreport required for this test.")
+        if dist.name == 'redhat':
+            if not sm.check_installed("sos") and not sm.install("sos"):
+                self.skip("Fail to install sos package required for this test.")
+        if distro.detect().name == 'SuSE':
+            self.skip('sosreport is not supported on SLES')
```
Comment (narasimhan-v) [inline generic/sosreport.py]: Can we handle other distros as well please ?

File: generic/sosreport.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    def run_cmd(self, cmd):
+        self.log.info("executing ============== %s =================" % cmd)
+        if process.system(cmd, ignore_status=True, sudo=True):
+            self.log.info("%s command failed" % cmd)
+            self.is_fail += 1
+        return
+
+    def setUp(self):
+        if "ppc" not in os.uname()[4]:
+            self.skip("supported only on Power platform")
```
Comment (narasimhan-v) [inline generic/sosreport.py]: I don't think sosreport is specific to power

File: generic/sosreport.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        if process.system_output("sosreport --batch --case-id=test123 -o cups | "
+                                 "grep kernel", shell=True, ignore_status=True):
+            self.is_fail += 1
+            self.log.info("--only-plugins option failed")
+        self.run_cmd("sosreport --batch --case-id=test123 -o dlm -k dlm.lockdump")
+        self.run_cmd("sosreport --batch --case-id=test123 -k filesys.dumpe2fs=on")
+        self.run_cmd("sosreport --batch --case-id=test123 --log-size=2000")
+        self.run_cmd("sosreport --batch --case-id=test123 -a")
+        self.run_cmd("sosreport --batch --case-id=test123 --all-logs")
+        self.run_cmd("sosreport --batch --case-id=test123 --build")
```
Comment (narasimhan-v) [inline generic/sosreport.py]: for opt in ["-o dlm -k dlm.lockdump", "-k filesys.dumpe2fs=on", ...]:
    self.run_cmd("sosreport --batch --case-id=test123 %s" % opt)

File: generic/sosreport.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        if not os.path.isdir(dir_name):
+            self.is_fail += 1
+            self.log.info("--build option failed")
+        self.run_cmd("sosreport --batch --case-id=test123 -v")
+        self.run_cmd("sosreport --batch --case-id=test123 --verify")
+        if process.system_output("sosreport --batch --case-id=test123 --quiet -e ntp,numa1", shell=True, ignore_status=True):
+            self.is_fail += 1
+            self.log.info("--quiet option failed")
+        self.run_cmd("sosreport --batch --case-id=test123 --debug")
+        if 'test123' not in process.system_output("sosreport --batch --ticket-number=test123  | "
```
Comment (narasimhan-v) [inline generic/sosreport.py]: Similar to what `run_cmd()` does for `process.system()`, you can have a function for `process.system_output()` as well, in all places below

**Comment by narasimhan-v** [comment]
Can we handle all these options as inputs from yaml file ?

File: generic/sosreport.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        self.run_cmd("sosreport --batch --case-id=test123 --chroot=never")
+        dir_name = process.system_output("sosreport --no-report --batch --build | "
+                                         "grep located | cut -d':' -f2", shell=True, ignore_status=True).strip()
+        sosreport_dir = os.path.join(dir_name, 'sos_reports')
+        if os.listdir(sosreport_dir) != []:
+            self.is_fail += 1
+            self.log.info("--no-report option failed")
+        self.run_cmd("sosreport --batch --case-id=test123 -s /home/")
+        process.run("rm -rf /tmp/sosreport*test123*")
+        process.run("rm -rf /root/sosreport*test123*")
```
Comment (nasastry) [inline generic/sosreport.py]: `rm -rf` is dangerous instead can you please use `mkdtemp - for temporary dir` and `mkstemp - for temporary file` and then delete them.

**Comment by Pavithra1602** [comment]
Raised new PR

**Pull request closed without merging**